Seq2SeqNMT/app/src/main/java/org/pytorch/demo/seq2seqnmt/MainActivity.java

```diff
@@ -0,0 +1,184 @@
+import android.widget.EditText;
+import android.widget.TextView;
+
+import androidx.appcompat.app.AppCompatActivity;
+
+import org.json.JSONException;
+import org.json.JSONObject;
+import org.pytorch.IValue;
+import org.pytorch.Module;
+import org.pytorch.PyTorchAndroid;
+import org.pytorch.Tensor;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.nio.FloatBuffer;
+import java.nio.LongBuffer;
+import java.util.ArrayList;
+
+public class MainActivity extends AppCompatActivity implements Runnable{
+    // to be consistent with the model inputs defined in seq2seq_nmt.py, based on
+    // https://pytorch.org/tutorials/intermediate/seq2seq_translation_tutorial.html
+    private final int HIDDEN_SIZE = 256;
+    private final int EOS_token = 1;
+    private final int MAX_LENGTH = 50;
+
+    private Module mModuleEncoder;
+    private Module mModuleDecoder;
+    private Tensor mInputTensor;
+    private LongBuffer mInputTensorBuffer;
+
+    private EditText mEditText;
+    private TextView mTextView;
+    private Button mButton;
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_main);
+
+        mButton = findViewById(R.id.btnTranslate);
+        mEditText = findViewById(R.id.etFrom);
+        mTextView = findViewById(R.id.tvTo);
+
+        mButton.setOnClickListener(new View.OnClickListener() {
+            public void onClick(View v) {
+            mButton.setEnabled(false);
+            Thread thread = new Thread(MainActivity.this);
+            thread.start();
+            }
+        });
+    }
+
+    private void showTranslationResult(String result) {
+        mTextView.setText(result);
+    }
+
+    public void run() {
+        final String result = translate(mEditText.getText().toString());
+        runOnUiThread(() -> {
+            showTranslationResult(result);
+            mButton.setEnabled(true);
+        });
+    }
+
+    private String translate(final String text) {
+        if (mModuleEncoder == null) {
+            mModuleEncoder = PyTorchAndroid.loadModuleFromAsset(getAssets(), "optimized_encoder_150k.pth");
+        }
+
+        String json;
+        JSONObject wrd2idx;
+        JSONObject idx2wrd;
+        try {
+            InputStream is = getAssets().open("target_idx2wrd.json");
+            int size = is.available();
+            byte[] buffer = new byte[size];
+            is.read(buffer);
+            is.close();
+
+            json = new String(buffer, "UTF-8");
+            idx2wrd = new JSONObject(json);
+
+            is = getAssets().open("source_wrd2idx.json");
+            size = is.available();
+            buffer = new byte[size];
+            is.read(buffer);
+            is.close();
+
+            json = new String(buffer, "UTF-8");
+            wrd2idx = new JSONObject(json);
+        } catch (JSONException | IOException e) {
+            e.printStackTrace();
```

Comment:
By default printStackTrace prints to System.err
it is not shown in logcat by default,
I think it is better to have it in logcat, smth like:

    android.util.Log.e(TAG, "Error reading json asset", e)